express-notes-api/index.js

```diff
@@ -0,0 +1,113 @@
+const express = require("express");
+const app = express();
+const fs = require("fs");
+const notesJson = require("./data.json");
+
+const writingFile = () => {};
+
+app.use(express.json());
+
+app.get("/api/notes", (req, res) => {
+  const gradesArray = [];
+  if (notesJson.notes) {
+    for (let notes in notesJson.notes) {
+      gradesArray.push(notesJson.notes[notes]);
+    }
+  } else if (!notesJson.notes) {
+    res.json([]);
+  }
+  res.json(gradesArray);
+});
+
+app.get("/api/notes/:id", (req, res) => {
+  let noteGet = {};
+
+  for (let notes in notesJson.notes) {
+    if (notesJson.notes[req.params.id]) {
+      noteGet = notesJson.notes[req.params.id];
+    } else if (req.params.id < 0) {
+      noteGet.error = "Id must be a positive integer";
+      res.status(400);
```

Comment:
A `400 Bad Request` response should be sent for _any_ `id` which isn't a positive integer:

---

![image](https://user-images.githubusercontent.com/16143771/144652935-3332eae3-521e-4900-aa2f-41e370e95c44.png)

---

![image](https://user-images.githubusercontent.com/16143771/144652970-9abb7256-e5ad-4ce7-899a-303eead44b0b.png)

Reply:
@GustavoAyala1 `id` validation is still needs to properly handle positive non-integers.